src/Compilers/VisualBasic/Portable/Operations/VisualBasicOperationFactory_Methods.vb

```diff
@@ -12,28 +12,127 @@ Namespace Microsoft.CodeAnalysis.Operations
+                               (binaryOperator.Left.Kind = BoundKind.Conversion AndAlso
+                                DirectCast(binaryOperator.Left, BoundConversion).Operand Is value.LeftOnTheRightOpt) Then
+                                Dim operatorKind = Helper.DeriveBinaryOperatorKind(binaryOperator.OperatorKind, binaryOperator.Left)
+                                Return (OperationKind.CompoundAssignment,
+                                        operatorKind,
+                                        IsChecked:=binaryOperator.Checked,
+                                        InConversion:=identityConversion,
+                                        outConversion,
+                                        RightNode:=binaryOperator.Right,
+                                        OperatorMethod:=Nothing)
+                            End If
+                        ElseIf rightConversion.Operand.Kind = BoundKind.UserDefinedConversion OrElse
+                               rightConversion.Operand.Kind = BoundKind.UserDefinedBinaryOperator Then
+
+                            ' In error scenarios, we can have a standard conversion on top of a UserDefinedBinaryOperator
+                            Dim rightOperatorBinary = If(TryCast(rightConversion.Operand, BoundUserDefinedBinaryOperator),
+                                                         TryCast(DirectCast(rightConversion.Operand, BoundUserDefinedConversion).Operand, BoundUserDefinedBinaryOperator))
+                            If rightOperatorBinary IsNot Nothing Then
+                                Dim inConversion As Conversion = identityConversion
+                                Dim operatorMethod As IMethodSymbol = Nothing
+                                If IsUserDefinedBinaryOperatorCompound(rightOperatorBinary, value.LeftOnTheRightOpt, inConversion, operatorMethod) Then
+                                    Dim operatorKind = Helper.DeriveBinaryOperatorKind(rightOperatorBinary.OperatorKind, leftOpt:=Nothing)
+                                    Return (OperationKind.CompoundAssignment,
+                                            operatorKind,
+                                            IsChecked:=rightOperatorBinary.Checked,
+                                            inConversion,
+                                            outConversion,
+                                            RightNode:=GetUserDefinedBinaryOperatorChildBoundNode(rightOperatorBinary, 1),
+                                            operatorMethod)
+                                End If
+                            End If
                         End If
                 End Select
             End If
 
-            Return OperationKind.SimpleAssignment
+            Return (OperationKind.SimpleAssignment, OperatorKind:=Nothing, IsChecked:=False, InConversion:=identityConversion, OutConversion:=identityConversion, RightNode:=Nothing, OperatorMethod:=Nothing)
```

Comment:
>Return (OperationKind.SimpleAssignment, OperatorKind:=Nothing, IsChecked:=False, InConversion:=identityConversion, OutConversion:=identityConversion, RightNode:=Nothing, OperatorMethod:=Nothing) [](start = 12, length = 194)

This should be put into the ```Else``` of the previous ```If```, otherwise it is not a SimpleAssignment.